### PR TITLE
=str #16602 Suppress dead letters

### PR DESCRIPTION
**Pull request by patriknw**
- akka 2.3.9
- also added missing `final` to case classes

### CODE REVIEW

File: akka-stream-testkit/src/test/scala/akka/stream/testkit/ScriptedTest.scala [hunk 1]
```diff
@@ -37,7 +37,7 @@ trait ScriptedTest extends Matchers {
     }
   }
 
-  case class Script[In, Out](
+  final case class Script[In, Out](
```
Comment (drewhk) [inline akka-stream-testkit/src/test/scala/akka/stream/testkit/ScriptedTest.scala]: Very nice!

File: akka-stream/src/main/scala/akka/stream/TimerTransformer.scala [hunk 2]
```diff
@@ -4,15 +4,15 @@
...
 import scala.collection.{ immutable, mutable }
 import scala.concurrent.duration.FiniteDuration
+import akka.actor.DeadLetterSuppression
 
 /**
  * Transformer with support for scheduling keyed (named) timer events.
  */
 // TODO: TimerTransformer is meant to be replaced; See https://github.com/akka/akka/issues/16410
-@deprecated("TimerTransformer is meant to be replaced; See https://github.com/akka/akka/issues/16410")
+@deprecated("TimerTransformer is meant to be replaced; See https://github.com/akka/akka/issues/16410", "1.0-M1")
```
Comment (drewhk) [inline akka-stream/src/main/scala/akka/stream/TimerTransformer.scala]: good one!

**Comment by drewhk** [comment]
LGTM

I hope we can eventually use abide and have rules for things like final case class.

File: akka-stream/src/main/scala/akka/stream/impl/TickPublisher.scala [hunk 3]
```diff
@@ -22,8 +21,8 @@ private[akka] object TickPublisher {
     Props(new TickPublisher(initialDelay, interval, tick, settings, cancelled)).withDispatcher(settings.dispatcher)
 
   object TickPublisherSubscription {
-    case object Cancel
-    case class RequestMore(elements: Long)
+    case object Cancel extends DeadLetterSuppression
```
Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/impl/TickPublisher.scala]: no final?

Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/impl/TickPublisher.scala]: `object`. nevermind.

File: project/AkkaBuild.scala [hunk 4]
```diff
@@ -1502,7 +1502,7 @@ object Dependencies {
 
     val reactiveStreamsVersion = System.getProperty("akka.build.reactiveStreamsVersion", "1.0.0.RC1")
 
-    val publishedAkkaVersion = "2.3.7"
+    val publishedAkkaVersion = "2.3.9"
```
Comment (ktoso) [inline project/AkkaBuild.scala]: :+1:

**Comment by ktoso** [comment]
LGTM, looking forward to having this merged :)

**Comment by patriknw** [comment]
Refs #16602

**Pull request merged**